src/coreclr/System.Private.CoreLib/src/System/Runtime/Loader/Assembly.cs

```diff
@@ -0,0 +1,57 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Runtime.Loader
+{
+    // BINDER_SPACE::Assembly represents a result of binding to an actual assembly (PEImage)
+    // It is basically a tuple of 1) physical assembly and 2) binder which created/owns this binding
+    // We also store whether it was bound using TPA list
+    internal sealed class BinderAssembly
+    {
+        // fields used by VM
+#pragma warning disable CA1823, 414, 169
+        private BinderAssemblyName m_assemblyName;
+
+        // The assembly object is rooted by native object.
+        // Reference native ALC to allow managed ALC object to be collected.
+        private IntPtr m_binder;
```

Comment:
This is a viable solution for now, but I don't know how it affects future design if the BINDER_SPACE::Assembly type is somehow attached with RuntimeAssembly.